freemocap/gui/qt/widgets/set_data_folder_dialog.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+import threading
+
+from PySide6.QtWidgets import QVBoxLayout, QDialog, QPushButton, QLabel, QHBoxLayout, QFileDialog
+
+from freemocap.gui.qt.utilities.save_and_load_gui_state import GuiState, save_gui_state
+from freemocap.system.paths_and_filenames.path_getters import get_gui_state_json_path
+
+
+class SetDataFolderDialog(QDialog):
+    def __init__(self, gui_state: GuiState, kill_thread_event: threading.Event, parent=None) -> None:
```

Comment:
just for good practice, move all this setup stuff into a method called `_initUI(self)` and then call that from the `__init__` method

this is apparently best practices for QT apps, and it does help clean things up a lot